src/app/code/community/FireGento/Logger/Helper/Data.php

```diff
@@ -170,7 +174,7 @@ public function addEventMetadata(&$event, $notAvailable = null, $enableBacktrace
             ->setStoreCode(Mage::app()->getStore()->getCode());
 
         // Add admin user data
-        if (Mage::app()->getStore()->isAdmin() && isset($_SESSION['admin'])) {
+        if (Mage::app()->getStore()->isAdmin() && isset($_SESSION) && isset($_SESSION['admin'])) {
```

Comment:
I'm not sure this extra check is actually necessary: although the documentation does not explicitly state this, performing an isset on an array index of an undefined variable returns `false` and it does not generate any warning:

```
$ php -r 'var_dump(isset($foo["bar"]));'
bool(false)
```

Reply:
From a warning perspective you are right that it is not necessary but in my experience this can cause a session to be initialized because it is accessing the `$_SESSION` global. In some contexts the session may not be initialized (e.g. a cron task).

Reply:
Ah yes, that sounds like a valid reason!